luigi_pipeline/lib/model/sv_mt_schema.py

```diff
@@ -0,0 +1,217 @@
+            and gene_col not in NON_GENE_PREDICTIONS
+        ]
+        mapped_genes = [
+            self.mt.info[gene_col].map(
+                lambda gene: hl.struct(**{
+                    GENE_SYMBOL: gene,
+                    GENE_ID: self._gene_id_mapping.get(gene, hl.missing(hl.tstr)),
+                    MAJOR_CONSEQUENCE: gene_col.replace(CONSEQ_PREDICTED_PREFIX, '', 1)
+                })
+            )
+            for gene_col in conseq_predicted_gene_cols
+        ]
+        return hl.filter(
+            hl.is_defined,
+            mapped_genes
+        ).flatmap(lambda x: x)
+
+    @row_annotation(fn_require=end_locus)
+    def xstop(self):
+        return variant_id.get_expr_for_xpos(self.mt.end_locus)
+
+    # rg37_locus is annotated by HailMatrixTableTask.add_37_coordinates
+
+    @row_annotation(fn_require=end_locus)
+    def rg37_locus_end(self):
+        return hl.or_missing(
+            self.mt.end_locus.position <= hl.literal(hl.get_reference('GRCh38').lengths)[self.mt.end_locus.contig],
+            hl.liftover(hl.locus(self.mt.end_locus.contig, self.mt.end_locus.position, reference_genome='GRCh38'), 'GRCh37'),
+        )
+
+    @row_annotation(name='svType')
+    def sv_type(self):
+        return self.sv_types()[0]
+
+    @row_annotation(name='transcriptConsequenceTerms', fn_require=[
+        sorted_transcript_consequences, sv_type,
+    ])
+    def transcript_consequence_terms(self):
+        return self.mt.sortedTranscriptConsequences.map(lambda x: x[MAJOR_CONSEQUENCE]).extend([self.mt.svType])
+
+    @row_annotation()
+    def sv_type_detail(self):
+        sv_types = self.sv_types()
+        return hl.if_else(
+            sv_types[0] == 'CPX',
+            self.mt.info.CPX_TYPE,
+            hl.or_missing(
+                (sv_types[0] == 'INS') & (hl.len(sv_types) > 1),
+                sv_types[1],
+            )
+        )
+
+    @row_annotation(name='geneIds', fn_require=sorted_transcript_consequences)
+    def gene_ids(self):
+        return hl.set(
+            self.mt.sortedTranscriptConsequences.filter(
+                lambda x: x[MAJOR_CONSEQUENCE] != 'NEAREST_TSS'
+            ).map(
+                lambda x: x[GENE_ID]
+            )
+        )
+
+    @row_annotation(name='variantId', disable_index=True)
+    def variant_id(self):
+        return self.mt.rsid
+
+    # NB: This is the "elasticsearch_mapping_id" used inside of export_table_to_elasticsearch.
+    @row_annotation(name='docId', disable_index=True)
+    def doc_id(self, max_length=512):
+        return self.mt.rsid[0: max_length]
+
+
+class SeqrSVGenotypesSchema(SeqrGenotypesSchema):
+
+    def _genotype_fields(self):
+        is_called = hl.is_defined(self.mt.GT)
+        return {
+            'sample_id': self.mt.s,
+            'gq': self.mt.GQ,
+            'cn': self.mt.RD_CN,
+            'num_alt': hl.if_else(is_called, self.mt.GT.n_alt_alleles(), -1)
```

Comment:
not very important, but it would be nice if this could inherit from the SeqrGenotypesSchema definition of num_alt

Reply:
👍